app/assets/javascripts/osm.js.erb

```
@@ -80,6 +80,11 @@ OSM = {
 
     const hash = OSM.parseHash(location.hash);
 
+    const geoURI = Object.fromEntries(new URLSearchParams(
+      params.geouri?.replaceAll(/[?;]/g, "&")?.replace(":", "=")
```

What you want to do requires zero regexes and replaces. Start with `new URL(params.geouri)`.

---

How does that help me with parsing semicolon-separated query parameters?

---

Also, the URL constructor throws a TypeError when no URI is given, instead of returning empty, which is what the current variant does without needing extra conditions

---

How many semicolon-separated parameters do you need to parse?

---

like `:`, `&` is possible inside geouri parameters https://datatracker.ietf.org/doc/html/rfc5870#section-3.3

---

> Are you going to look at anything past the first semicolon?

In RFC 5870 uncp comes after crsp, so to parse `geo:57.6247,-3.6845;crs=wgs84;u=100`, I need to.
> like `:`, `&` is possible inside geouri parameters

Well then `new URL()` isn't going to help me either.

---

`new URL()` is going to help you to get `z` and to get everything between `geo:` and `?` without writing things like `slice(4)`.

Do you plan to show the uncertainty? Because if not, maybe you don't need `u`. And if you need `u`, surely there's a simpler regexp just for `u` than `/;([\w-]+)=([\w-.!~*'()[\]:&+$%]+)/g`.

---

So do you want me to follow RFC 5870 or not?
https://github.com/openstreetmap/openstreetmap-website/pull/5736#discussion_r1979882902 indicates yes, https://github.com/openstreetmap/openstreetmap-website/pull/5736#discussion_r1976402137 indicates no

---

If you don't need `u`, you don't need any regexps.

If you need `u`, that parameter can't have `&` inside and you can have a simpler regexp. But do you need `u`? Are you going to show the uncertainty?

If you want to parse *every* parameter, you'll need a complicated regexp. What do you need other parameters for?

`z` is not a RFC 5870 parameter.